src/mono/mono/mini/interp/transform-simd.c

```diff
@@ -455,6 +468,25 @@ emit_sri_vector128 (TransformData *td, MonoMethod *cmethod, MonoMethodSignature
 			simd_intrins = INTERP_SIMD_INTRINSIC_V128_BITCAST;
 			break;
 		}
+		case SN_AsVector:
+		case SN_AsVector128:
+		case SN_AsVector4: {
+			if (!is_element_type_primitive (csignature->ret) || !is_element_type_primitive (csignature->params [0]))
```

Comment:
This will allow <bool> through, won't it?

Reply:
No, it calls `MONO_TYPE_IS_VECTOR_PRIMITIVE` which explicitly limits the range to `((!m_type_is_byref ((t)) && ((((t)->type >= MONO_TYPE_I1 && (t)->type <= MONO_TYPE_R8) || ((t)->type >= MONO_TYPE_I && (t)->type <= MONO_TYPE_U)))))`

and therefore excludes `bool`, `char`, and others.

Reply:
OK, I dug through and the name `is_element_type_primitive` is misleading - it's limited to vector element types.

Reply:
> the name is_element_type_primitive is misleading

👍, in RyuJIT we call the API `isSupportedBaseType` instead (although `isSupportedSimdBaseType` would've been even better). The `is_element_type_primitive` name is the one that `mono-mini` had used and so it just propagated here into the interpreter as well.